crates/net/network/src/manager.rs

```diff
@@ -919,3 +928,12 @@ pub enum NetworkEvent {
     /// Event emitted when a new peer is removed
     PeerRemoved(PeerId),
 }
+
+#[derive(Debug, Clone)]
+pub enum DiscoveredEvent{
+    EventQueued{
+        peer_id: PeerId,
+        socket_addr: SocketAddr,
+        fork_id: Option<ForkId>
+    }
```

Comment:
we don't need this

Reply:
If we remove this , then how will we encapsulate `peer_id`, `socket_addr`, and `fork_id`?

Reply:
oh I get it now, very good point!

will give it another look, I think this is good now, pending the listner cleanup,

thanks